email/provider_test.go

```diff
@@ -12,6 +12,9 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+// Global Testing Variables
+var testRetries int = 0
```

Comment:
```suggestion
var sendMailInvocations int = 0
```

Very nit, but "retry" seems to imply that there was already a "try"...